### ISSUE TITLE
'detach' flag is 'True' when calling hook_outputs() in class DynamicUnetWide

### ISSUE THREAD

**Issue by macul99** [jantic/DeOldify#116]
Hi,

Your model is amazing and thank you very much for publishing the code. I read your code and have one question to clarify:

From my understanding, the hook_outputs() is used to generate skip connections when DynamcUnetWide instance is created. However, the default value for 'detach' flag in fastai library is 'True':

`def hook_outputs(modules:Collection[nn.Module], detach:bool=True, grad:bool=False)->Hooks:
    return Hooks(modules, _hook_inner, detach=detach, is_forward=not grad)`

In this case, the back-propagation will stop at the skip-connections. Is it intended to do so or my understanding is wrong?

Thank you very much!

**Comment by jantic** [comment]
Interesting finding!  Yeah that was surprising to me too.... This actually came from fastai's DynamicUnet.  I just modified that a bit.  It still does this same thing.

The good news is that having this line:

> layers = [encoder, batchnorm_2d(ni), nn.ReLU(), middle_conv]

Means that there's still some learning going on with the backbone because backpropogation will go back through these layers unimpeded.  I can verify that weights are changing on Tensorboard for the resnet backbone as I train a model now.

To complicate matters:  The detach is apparently not happening in the original implementation of Unet:  https://github.com/fastai/fastai/blob/master/old/fastai/models/unet.py

So yeah I'm guessing it's not intended but I'm not 100% sure, and I'm not sure just how big of a difference it makes.... I'll try a run that doesn't detach now and see what happens....

**Comment by jantic** [comment]
It'll take a while to get definitive results (a few days I think).  But do know I'm taking a look. Thanks again.

**Comment by macul99** [comment]
Thank you very much! Looking forward to hear from you the findings!

**Comment by jantic** [comment]
Sorry for the long wait.  So yes- it seems like setting detach=False is beneficial.  It isn't hugely beneficial which might be surprising but regardless it makes sense to do it and it works. I've committed this change.  Thanks!

**Comment by macul99** [comment]
Thx for the update!
